syntax = "proto3";
package rl;

import "common.proto";

message SampleRequest {
  string experiment_id = 1;
  int32 batch_size = 2;
}

service BufferService {

  // Actors push experience here
  rpc PushTransition(Transition) returns (Status);

  // Learner samples minibatches
  rpc SampleBatch(SampleRequest) returns (Batch);

  // Optional: get buffer stats
  rpc GetBufferSize(ExperimentConfig) returns (Metrics);
}
